### PR TITLE
Add capability to delete block from Python SDK

### PR DESCRIPTION
**Pull request by rito-sixt**
closes https://github.com/PrefectHQ/prefect/issues/9755 

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

**Review by chrisguidry — APPROVED** [review]
That's a nice addition and a nice clear test, thanks!

File: tests/blocks/test_core.py [hunk 1]
```diff
@@ -2477,3 +2477,30 @@ class Baz_Alias(Block):
...
+            a: str
+            b: str
+
+        return NewBlock
+
+    async def test_delete_block(self, NewBlock):
+        new_block = NewBlock(a="foo", b="bar")
+        new_block_name = "my-block"
+        await new_block.save(new_block_name)
+
```
Comment (chrisguidry) [inline tests/blocks/test_core.py:2498]:
```suggestion
        await new_block.save(new_block_name)
        await new_block.load(new_block_name)

```

Not terribly important, just a stylistic thing.  In tests, I often like to show that something would work before the operation I'm testing, then show how it works after my operation.  So in this case, I can `.load` it before, then I `.delete` it, so afterward I can't `.load` it again.  Your test is totally fine without it, just a style thing.

File: tests/blocks/test_core.py [hunk 2]
```diff
@@ -2477,3 +2477,30 @@ class Baz_Alias(Block):
 
         # new local schema ID should now be saved to Prefect
         assert updated_schema_id == new_schema_id
+
+
+class TestDeleteBlock:
```
Comment (desertaxle) [inline tests/blocks/test_core.py:2482]: Could you also add a test covering `.delete()` when used with the base `Block` class? (e.g. `Block.delete("newblock/my-block")`)

**Comment by rito-sixt** [comment]
> That's a nice addition and a nice clear test, thanks!

Thanks for the encouraging remark @chrisguidry! Means a lot 🙌

File: tests/blocks/test_core.py (hunk 1 above)
Comment (rito-sixt) [reply tests/blocks/test_core.py:2498]: Thanks for the input @chrisguidry , and completely agreed that it makes the intent of the test much clearer. Added the same now.

File: tests/blocks/test_core.py (hunk 2 above)
Comment (rito-sixt) [reply tests/blocks/test_core.py:2482]: Great catch @desertaxle ! Added the test now! Thanks a ton for the input! 🙌

**Review by chrisguidry — APPROVED** [review]

**Review by desertaxle — APPROVED** [review]
LGTM! Thanks so much for the awesome contribution!

**Comment by rito-sixt** [comment]
> LGTM! Thanks so much for the awesome contribution!

Thanks a ton for the kind words! 🙌

**Pull request merged**